flaml/automl.py

```diff
@@ -2447,6 +2460,7 @@ def custom_metric(
         eval_method = self._decide_eval_method(eval_method, time_budget)
         self._state.eval_method = eval_method
         logger.info("Evaluation method: {}".format(eval_method))
+        self._state.cv_strategy = cv_strategy
```

Comment:
Change this line after you change the constructor of `AutoML`. Refer to line 2433 for example.